ebean-core/src/main/java/io/ebeaninternal/server/persist/DefaultPersister.java

```diff
@@ -417,7 +418,7 @@ public void update(EntityBean entityBean, Transaction t) {
       } else {
         update(req);
       }
-
+      req.resetDepth();
```

Comment:
Reset the depth after each top level insert and update.  This means that all our top level requests at depth zero are treated in the order that they are saved().